addons/website_payment/static/src/snippets/s_donation/options.js

```diff
@@ -0,0 +1,249 @@
+/** @odoo-module **/
+
+import {_t, qweb} from 'web.core';
+import options from 'web_editor.snippets.options';
+
+options.registry.Donation = options.Class.extend({
+    xmlDependencies: ['/website_payment/static/src/snippets/s_donation/000.xml'],
+
+    /**
+     * @override
+     */
+    onBuilt() {
+        this.defaultDescription = _t("Add here a description");
+        this._rebuildPrefilledOptions();
+        this._refreshPublicWidgets(); // we need it to rebuild currencies after the rebuild of the options
+        return this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    cleanForSave() {
+        this.$target.find('.s_donation_currency').remove();
+        if (!this.$target[0].dataset.descriptions) {
+            this._updateDescriptions();
+        }
```

Comment:
Not really, it is because when the user disable the option "Descriptions", the input hidden are still in the DOM so they won't be lost if the user wants to activate the option later when editing the page. So the input hidden are removed only when the user leave the page if the option "Descriptions" is disable